src/lib.rs

```diff
@@ -139,12 +139,58 @@ pub fn select_coin_lowestlarger(
 }
 
 /// Perform Coinselection via First-In-First-Out algorithm.
+
 /// Return NoSolutionFound, if no solution exists.
+
 pub fn select_coin_fifo(
     inputs: &[OutputGroup],
     options: CoinSelectionOpt,
 ) -> Result<SelectionOutput, SelectionError> {
-    unimplemented!()
+    let mut accumulated_value: u64 = 0;
+    let mut accumulated_weight: u32 = 0;
+    let mut selected_inputs: Vec<usize> = Vec::new();
+    let mut estimated_fees: u64 = 0;
+
+    // Sorting the inputs vector based on creation_sequence
+
+    let mut sorted_inputs: Vec<_> = inputs.iter().enumerate().collect();
+
+    sorted_inputs.sort_by_key(|(_, a)| a.creation_sequence);
+
+    for (index, inputs) in sorted_inputs {
+        estimated_fees = calculate_fee(accumulated_weight, options.target_feerate);
+        if accumulated_value
+            >= (options.target_value
+                + options.target_value
+                + estimated_fees
+                + options.min_drain_value)
```

Comment:
Have to make sure that the accumulated_value has sufficient fee also, `estimated_fee.max(options.min_absolute_fee)`

Reply:
Right.